presto-spi/src/main/java/io/prestosql/spi/type/VarcharType.java

```diff
@@ -261,4 +262,10 @@ private static long comparisonOperator(@BlockPosition Block leftBlock, @BlockInd
         int rightLength = rightBlock.getSliceLength(rightPosition);
         return leftBlock.compareTo(leftPosition, 0, leftLength, rightBlock, rightPosition, 0, rightLength);
     }
+
+    public boolean narrowerThan(VarcharType other)
```

Comment:
Let's not add this to SPI. We should add it when it is commonly needed.
This can be just a helper method next to `createCoercer`